### PR TITLE
libgit: implement the delete repo RPC call

### PR DESCRIPTION
**Pull request by strib**
And vendor in the protocol changes for it.

Depends on keybase/client#8478, will revendor once that's merged.

Issue: KBFS-2435

### CODE REVIEW

File: libgit/repo.go [hunk 1]
```diff
@@ -262,3 +266,52 @@ func CreateRepoAndID(
 	}
 	return id, err
 }
+
+// DeleteRepo "deletes" the given repo in the given TLF.  RIght now it
```
Comment (jzila) [inline libgit/repo.go]: s/RIght/Right

File: libgit/repo.go [hunk 2]
```diff
@@ -262,3 +266,52 @@ func CreateRepoAndID(
...
+
+	deletedReposNode, err := lookupOrCreateDir(
+		ctx, config, repoNode, kbfsDeletedReposDir)
+	if err != nil {
+		return err
+	}
+
+	// For now, just rename the repo out of the way, using the uniq
+	// ID.  TODO(KBFS-2442): periodically delete old-enough repos from
+	// `kbfsDeletedReposDir`.
```
Comment (jzila) [inline libgit/repo.go]: So `kbfsDeletedReposDir` is a staging area for deleted repos? Meaning anything there is good to be deleted?

File: libgit/repo.go [hunk 3]
```diff
@@ -262,3 +266,52 @@ func CreateRepoAndID(
...
+func DeleteRepo(
+	ctx context.Context, config libkbfs.Config, tlfHandle *libkbfs.TlfHandle,
+	repoName string) error {
+	// Create a unique ID using the verifying key and the `config`
+	// object, which should be unique to each call in practice.
+	session, err := config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	uniqID := fmt.Sprintf("%s-%p", session.VerifyingKey.String(), config)
```
Comment (jzila) [inline libgit/repo.go]: I don't love using the pointer value of `config` as the unique ID. Is there a reason not to use `kbfscrypto.RandRead`?

File: libgit/repo.go (hunk 2 above)
Comment (strib) [reply libgit/repo.go]: Yes, though it's probably best to wait until the renamed repo has been deleted at least a few minutes, to make sure any concurrent writers have finished pushing to it.

File: libgit/repo.go (hunk 3 above)
Comment (strib) [reply libgit/repo.go]: It was easier, and it's guaranteed to be unique for this call.  Technically a random number isn't guaranteed to be unique.  Why don't you like it?

Comment (jzila) [reply libgit/repo.go]: I don't understand how a pointer is guaranteed to be unique.

Comment (strib) [reply libgit/repo.go]: It only needs to be unique "during the lifetime of this call", as the function comment explains.  A new config is passed into each of these calls, and therefore any calls happening at the same time will have unique pointers.

Comment (jzila) [reply libgit/repo.go]: But the new config's pointer can be the same as the old one...

Comment (jzila) [reply libgit/repo.go]: Oh nvmind I follow, different configs _at the same time_

Comment (strib) [reply libgit/repo.go]: Yeah that's what I was thinking, but you're right.  By including it in the name, it needs to be unique for all time, or at least until the next delete run happens.  I'll fix it soonish.

**Comment by strib** [comment]
@jzila: fixed the uniqueness issue in 6229826.  If you agree, then this can be merged once keybase/client#8478 is approved and merged.

**Pull request merged**